internal/storedconfig/stored_config.go

```diff
@@ -20,73 +20,31 @@ import (
 	"strings"
 
 	"github.com/rs/zerolog"
+
 	"github.com/snyk/go-application-framework/pkg/configuration"
 
 	"github.com/snyk/snyk-ls/internal/types"
 )
 
-// GetOrCreateFolderConfig queries git for the folder config of the given path
+// GetOrCreateFolderConfig gets folder config from storage and merges it with Git data
 func GetOrCreateFolderConfig(conf configuration.Configuration, path types.FilePath, logger *zerolog.Logger) (*types.FolderConfig, error) {
-	folderConfig, err := folderConfigFromStorage(conf, path, nil)
-	if err != nil {
-		return nil, err
-	}
+	l := logger.With().Str("method", "GetOrCreateFolderConfig").Logger()
 
-	gitFolderConfig, err := getFromGit(path)
+	folderConfig, err := folderConfigFromStorage(conf, path, &l)
 	if err != nil {
-		return folderConfig, nil
+		return nil, err
 	}
 
-	if folderConfig != nil && gitFolderConfig != nil {
-		// the previous git configuration takes precedence
-		folderConfig = mergeFolderConfigs(gitFolderConfig, folderConfig)
-	}
+	enrichFromGit(&l, folderConfig)
```

Comment:
Hm. shouldn't that be assigned back? Or is it updating the pointer parameter? In any case, it would be cleaner, to return the enriched value and assign it to `folderConfig`.